src/mapreduce.jl

```diff
@@ -176,6 +176,13 @@ end
 
 @inline reduce(op, a::StaticArray; dims=:, kw...) = _reduce(op, a, dims, kw.data)
 
+# disambiguation
+reduce(::typeof(vcat), A::StaticVector{<:Any,<:AbstractVecOrMat}) =
+    Base._typed_vcat(mapreduce(eltype, promote_type, A), A)
```

Comment:
It's worthwhile to get the ambiguities down to zero and keep them there, so I had a closer look. Method specificity is a complex heuristic so it can be hard to figure out which methods to write but in this case it's because of the `Any` in `StaticVector`. Amusingly the following broader-looking signature seems to fix it:

```julia
reduce(::typeof(vcat), A::StaticArray{<:Tuple,<:AbstractVecOrMat})
```

Also, perhaps we should dispatch to our own internal `_reduce` here rather than `Base._typed_vcat`?

Reply:
That's quite interesting indeed. I'll change the signature then.

> Also, perhaps we should dispatch to our own internal `_reduce` here rather than `Base._typed_vcat`?

That's a good question. `Base._typed_vcat` preallocates output for the whole array that will be returned using `similar` with calculated new size. That should be faster than our `_reduce` for sized arrays of normal `Array`s but it doesn't return a static array when it gets a static array of static arrays. So I think we should have two methods here: one for static arrays of static arrays that calls our `_reduce` and one for everything else. Does it sound reasonable?

Reply:
It's a bit of a weird special case, given that we don't have any other special cases of `reduce(op, a::StaticArray; dims=:, kw...)`; it makes me think that maybe this signature should have been tighter in some way (though I'm not sure quite how).

Anyway, yes, let's do as you suggest for now.

Reply:
I've made these changes and now `StaticArrays` has two ambiguities less :tada: .